### ISSUE TITLE
Couple of issues with redwood

### ISSUE THREAD

**Issue by jonniebigodes** [redwoodjs/redwood#305]
Following the tutorial i've encountered a couple of issues that are somewhat of concern. One needs attention but not so immediate as the other.

Starting from the one least problematic one.

- After following the tutorial up until the [layouts section](https://redwoodjs.com/tutorial/layouts) all is good and everything is working properly. But when i reach this section (and i caught this by mere chance) when i spin up the development server with `yarn redwood dev` after going through the steps detailed in that section, my system is opening up ms edge and it displays this to me:

![redwood1](https://user-images.githubusercontent.com/22988955/77210928-226a9500-6afa-11ea-9c6d-85e6b2170325.png)

And the developer tools in ms edge
![redwood2](https://user-images.githubusercontent.com/22988955/77210948-30b8b100-6afa-11ea-8409-d7e75438a9ab.png)

As a mere curiosity i opened up firefox and it's also showing the same as you can see below:

![redwood3](https://user-images.githubusercontent.com/22988955/77211001-6067b900-6afa-11ea-9e66-8652523fce5d.png)

Now onto the most problematic one:

- Taking the above into consideration, i went with chrome and all was working fine until the [saving-data section](https://redwoodjs.com/tutorial/saving-data) 

After going through the steps in the said section up until the `GraphQL Playground` part and taking into consideration #284 (i tested it with both cmd and the windows terminal preview) workarounds to get it to work. I get the following:

```bash
yarn run v1.21.1
$ C:\Users\MyUser\Myfolder\redwood-blog-test\node_modules\.bin\redwood dev
$ C:\Users\MyUser\Myfolder\redwood-blog-test\node_modules\.bin\dev-server
$ C:\Users\MyUser\Myfolder\redwood-blog-test\node_modules\.bin\prisma2 generate --watch
$ C:\Users\MyUser\Myfolder\redwood-blog-test\node_modules\.bin\webpack-dev-server --config ../node_modules/@redwoodjs/core/config/webpack.development.js
22:35:13 api |
22:35:13 api | C:\Users\MyUser\Myfolder\redwood-blog-test\node_modules\graphql\type\validate.js:71
22:35:13 api |     throw new Error(errors.map(function (error) {
22:35:13 api |           ^
22:35:13 api | Error: The type of Mutation.createContact(input:) must be Input Type but got: ContactInput!.
22:35:13 api |     at assertValidSchema (C:\Users\MyUser\Myfolder\redwood-blog-test\node_modules\graphql\type\validate.js:71:11)
22:35:13 api |     at assertValidExecutionArguments (C:\Users\MyUser\Myfolder\redwood-blog-test\node_modules\graphql\execution\execute.js:136:35)
22:35:13 api |     at executeImpl (C:\Users\MyUser\Myfolder\redwood-blog-test\node_modules\graphql\execution\execute.js:86:3)
22:35:13 api |     at Object.execute (C:\Users\MyUser\Myfolder\redwood-blog-test\node_modules\graphql\execution\execute.js:64:63)
22:35:13 api |     at Object.generateSchemaHash (C:\Users\MyUser\Myfolder\redwood-blog-test\node_modules\apollo-server-core\src\utils\schemaHash.ts:11:18)
22:35:13 api |     at ApolloServer.generateSchemaDerivedData (C:\Users\MyUser\Myfolder\redwood-blog-test\node_modules\apollo-server-core\src\ApolloServer.ts:496:24)
22:35:13 api |     at new ApolloServerBase (C:\Users\MyUser\Myfolder\redwood-blog-test\node_modules\apollo-server-core\src\ApolloServer.ts:369:32)
22:35:13 api |     at new ApolloServer (C:\Users\MyUser\Myfolder\redwood-blog-test\node_modules\apollo-server-lambda\src\ApolloServer.ts:38:5)
22:35:13 api |     at createGraphQLHandler (C:\Users\MyUser\Myfolder\redwood-blog-test\node_modules\@redwoodjs\api\src\graphQLServer.ts:48:19)
22:35:13 api |     at Object.<anonymous> (C:\Users\MyUser\Myfolder\redwood-blog-test\ap
i\src\functions\graphql.js:31:47)
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
22:35:13 api | cd C:\Users\MyUser\Myfolder\redwood-blog-test\api && yarn dev-server exited with code 1
22:35:13 api | cd C:\Users\MyUser\Myfolder\redwood-blog-test\api && yarn dev-server restarted
```

I've gone over this and see if i made some mistakes and the code matches the documentation.

Feel free to provide feedback

**Comment by thedavidprice** [comment]
Hi @jonniebigodes Thanks for checking in here and we’ll hopefully have you up an running soon!

Just to make sure we understand, **at the Layouts section**, Chrome is working fine, but you’re seeing errors for Edge and Firefox, correct? (Note: we are currenlty working on a work-around for a known FF bug.)

But in the **“Saving Data” section**, things blow up across the board, correct?

### Versions and Reproduction
I see your Yarn version is v1.21.1. Can you confirm your Node, OS, and RedwoodJS package versions?

Also, just to make sure, have you stopped the dev server and re-tried:
`yarn rw db save`
`yarn rw db up`

Lastly, if you have a chance to put the code in a public repo, that might help us diagnose.

**Comment by jonniebigodes** [comment]
> Just to make sure we understand, at the Layouts section, Chrome is working fine, but you’re seeing errors for Edge and Firefox, correct? (Note: we are currenlty working on a work-around for a known FF bug.)

That is correct. Hence the screenshots.

> But in the “Saving Data” section, things blow up across the board, correct?

Yes that is correct.

Running Node v12.15.0 on Windows 10 1903 with `@redwoodjs/core` version `0.2.5`


> Also, just to make sure, have you stopped the dev server and re-tried:
> yarn rw db save
> yarn rw db up

Yes as per mentioning in the documentation, also I deleted the database and migrations and re ran the commands above to see if it might be related to a issue with the database/migrations and same error.

> Lastly, if you have a chance to put the code in a public repo, that might help us diagnose.

My code is [here](https://github.com/jonniebigodes/code_redwood_issue_305) i left the database and migrations. Also the database is quasi empty as when i commited it i just had finished recreating it.

Feel free to provide feedback

**Comment by paustint** [comment]
@jonniebigodes - I am not sure if this is the issue or related, but I found that going through the tutorial `contacts.sdl.js` was generated slightly differently from a code example shown in the tutorial.

If you replaced the generated code with what the tutorial showed you on this page, https://redwoodjs.com/tutorial/saving-data, then it could have possibly caused the issue.

```diff
-  type ContactInput {
+  input ContactInput {
```

**The generated code is this:**
```
export const schema = gql`
  type Contact {
    id: Int!
    name: String!
    email: String!
    message: String!
    createdAt: DateTime!
  }

  type Query {
    contacts: [Contact]
  }

  input ContactInput {
    name: String
    email: String
    message: String
  }

  type Mutation {
    createContact(input: ContactInput!): Contact
  }
`;

```

**The tutorial shows this as a code example:**
```
export const schema = gql`
  type Contact {
    id: Int!
    name: String!
    email: String!
    message: String!
    createdAt: DateTime!
  }

  type Query {
    contacts: [Contact]
  }

  type ContactInput {
    name: String
    email: String
    message: String
  }

  type Mutation {
    createContact(input: ContactInput!): Contact
  }
`;
```

**Comment by thedavidprice** [comment]
@jonniebigodes I looked through your reprodution repo — I want to confirm what @paustint wrote above. It seems the tutorial reference code incorrectly swapped the “input” for “type”. (Thanks for the PR @paustint!) However, when the code is generated it is set correctly as “input”. Confusing indeed. 

I made the change locally (replaced “type” with “input" and can confirm things run correctly.

But there could be other issues still at play so please do confirm on your end.

And sorry for the frustration here. Appreciate you plugging in and sticking with us!

**Comment by jonniebigodes** [comment]
@thedavidprice it seems so, i checked earlier what @paustint mentioned and it's working correctly on my end as well. The only issue was because i "fumble fingered" the date in the db but suprisingly enough it didn't break nothing. The loading went through, showed the error regarding the date and displayed the information correctly. Also no need to apologize, glad that i was able to help out. 

Regarding the Firefox bug i saw that it's a upstream bug on their end so for now i'm considering it resolved as they fix on their end. And on edge it might be that some polyfills might be required to solve the issue.

**Comment by thedavidprice** [comment]
Great to hear.

**And re: Firefox,** Peter has been doing some great work fixing issue on both the Firefox and Windows side of things. Here are upcoming fixes available in the next release, which will be v0.3:
https://github.com/redwoodjs/redwood/releases/tag/untagged-85ca562a1e413d4479d0

And more to come today before this is published. 

Closing this out. Thanks again!
